src/strategies/immutable-x/index.ts

```diff
@@ -0,0 +1,127 @@
+import { BigNumber, BigNumberish } from '@ethersproject/bignumber';
+
+import { Multicaller } from '../../utils';
+import { StaticJsonRpcProvider } from '@ethersproject/providers';
+import fetch from 'cross-fetch';
+import { formatUnits } from '@ethersproject/units';
+
+export const author = 'immutable';
+export const version = '1.0.0';
+
+export const name = 'immutable-x';
+
+const snapshotPath = '/v1/snapshots/balances';
+
+const networkMapping = {
+    1: 'https://api.x.immutable.com',
+    3: 'https://api.ropsten.x.immutable.com'
+};
+
+const defaultPageSize = 100;
+
+const abi = ['function balanceOf(address account) external view returns (uint256)'];
+
+interface Response {
+    records: Score[];
+    cursor: string;
+}
+
+interface Score {
+    ether_key: string;
+    balance: string;
+}
+
+interface Options {
+    address: string;
+    decimals: number;
+    pageSize?: number;
+}
+
+export async function strategy(
+    _space: unknown,
+    network: string,
+    provider: StaticJsonRpcProvider,
+    addresses: string[],
+    options: Options,
+    block: number | string = 'latest'
+): Promise<Record<string, number>> {
+    return combineBalanceScores([
+        await getL2Balances(network, options, addresses, block),
+        await getL1Balances(network, provider, options, addresses, block)
+    ]);
+}
+
+async function getL1Balances(
+    network: string,
+    provider: StaticJsonRpcProvider,
+    options: Options,
+    addresses: string[],
+    block: number | string
+): Promise<Record<string, number>> {
+    const multi = new Multicaller(network, provider, abi, { blockTag: block });
+    addresses.forEach((address: string) => multi.call(address, options.address, 'balanceOf', [address]));
+    const result: Record<string, BigNumberish> = await multi.execute();
+    return mapL1Response(result, options);
+}
+
+async function getL2Balances(network: string, options: Options, addresses: string[], block: number | string): Promise<Record<string, number>> {
```

Comment:
@ChaituVR This strategy adds L2 and L1 balances. Did you mean I should leverage another strategy within this one for L1 only?

Reply:
Hi @sergerad, No, I mean in this case you can use multichain strategy and pass two `erc20-balance-of` strategies into it 

More details here: https://snapshot.org/#/strategy/multichain
Play with it more and let me know if it multichain strategy doesn't work for you

Reply:
The voting strategy "multichain" would only work if the chain is supported. If you want to get voting power from StarkEx for example I would recommend keep this approach.

Reply:
I have used `erc20-balance-of` strategy for L1 in this PR 👍

Reply:
@bonustrack PR is labelled with `changes requested` but my interpretation of your comment was to keep the current impl

Reply:
@sergerad Can you revert changes from `yarn.lock` file please, everything else looks good

Reply:
@ChaituVR done, thanks